### ISSUE TITLE
Request Queue for unopened requests

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#20]
As an intake analyst, I want to see a list of FOI requests so that I can understand the status of requests and action requests.

**Assumption**	
Treat request as being unopened
No editing information on the Queue - edits will be at single view of a request.
Will need a custom page created in Forms flow. Include Get from API. (TBD based on technical convos)
API work is in #27 to capture webform request.

**Acceptance Criteria**	

**Scenario 1: Data in each row for a request**
* GIVEN an intake analyst has logged in successfully
* WHEN the intake analyst is on the Queue
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/657932451
* THEN for each request the analyst will see applicant last name and first name, request type (general or personal), file number (blank for 'Unopened' request), current status (Unopened), Assigned to (blank for 'Unopened' requests), date submitted, days left 

**Scenario 2: Order of requests on Queue**
* GIVEN an intake analyst has logged in successfully
* WHEN the intake analyst is on the Queue
* THEN the requests will  sorted by Unopen at top (by date submitted from oldest to newest)

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
See designs in Scenarios

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.     [ ] Passes developer unit tests
2.    [x] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[x] Developer to list Config changes/ Update documents and designs
10.	[x] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
@arielleandrews, @m-prodan, @JHarrietha-AOT please review the Scenarios above. 

Notes:.
 
1. How this functionality works will change once @JHarrietha-AOT as done the designs, but put in examples (e.g radio button???) as placeholders for now.

2, this is a lot in one story so after the review I will break the Sort and Filter functionality out into a separate ticket.

cc @abin-aot, @mpilchar

**Comment by JHarrietha-AOT** [comment]
@m-prodan @arielleandrews Wondering if it makes more sense to have the unopened queue items in order from oldest - newest? Looking at my notes I think that is the order Intact would work through them in.

**Comment by m-prodan** [comment]
Agree @JHarrietha-AOT - Intake mentioned they prioritize their work by date submitted, working in an order from oldest to newest - that would be more helpful as a default sort for them.

In addition, I know we heard that they don't need or want to see closed requests, we could have a "hide closed requests" on as a default for the purposes of a queue. Otherwise, everything else looks good to me. There's a few spots where we're missing a user (scenario 3-5), but that's an easy fix.

**Comment by arielleandrews** [comment]
All Scenarios: if the idea of the queue is to see newly received/active requests, I don't see including Closed requests here, so request status options for now would be Unopened (or Received) and Opened only. If Closed files need to be included for simplicity for the MVP that works, but longer term Intake analysts likely won't want to see Closed files

Scenario 3: This is a bit unclear to me, are we talking about the sort order here? Requests will be sorted by selected column first, then oldest to newest as the second level sort, and then open requests oldest to newest as the third level sort? @JHarrietha-AOT is right that Intake would prefer oldest to newest as they will work on the oldest ones first

**Comment by JHarrietha-AOT** [comment]
@m-prodan I like the idea of having open (closed to intact) requests as a toggle on or off feature. It can be defaulted to off, but if an Intact Analyst is processing a payment they would need to add that to the correspondence log of an open request. But I don't think correspondence will be in the MVP so maybe the toggle is a feature we wait on. @lmullane what do you think?

@arielleandrews My understanding is that scenario 3 is sorting by the columns heading. (eg. request 3) I wonder if it would be more clear if we remove the first AND (AND requests will be ordered by the newest to the oldest number), as that is covered in scenario 2 in the default set-up. Would that make it more clear?

**Comment by arielleandrews** [comment]
Good point Jackie! I wonder then if we need the ANDs at all for Scenario 3 at this point? Thoughts team?

**Comment by JHarrietha-AOT** [comment]
@arielleandrews If we are looking at removing the opened requests from the intact queue for the MVP then we can remove the second AND as well. @lmullane do you think we should include open requests for the MVP or just focus on the process of opening a request?

**Comment by lmullane** [comment]
Great feedback everyone.

I have updated Scenario 1 to show unopened requests from oldest to newest.

I have also axed 'Closed' requests which will no longer appear. In future we may want an archive but that sounds like it would be better presented by the search for applicant name or file number. Should the filters in scenario 6 and 7 show 'Closed' requests (note, these would not include historical requests in AXIS) at this time.

I was going to pull the filter scenarios out of this story as they may not be part of MVP.

@arielleandrews, Scenario 3 would be the default sort when you first visited the page. The sort would enable you to change it back to the default in case you had sorted it by something else. @JHarrietha-AOT, welcome your design lens on how to return to the default! If we need a Scenario 3 it might be totally different:)

@JHarrietha-AOT, I'm leaning towards including 'Open' requests.  My understanding is the intake analyst assigns the task to their supervisor who likey will use the same Queue and will likely want full visibility on both Unopen and Open requests. I also think it may be a bit weird for the request to just disappear when you Open it. What do you think?

**Comment by lmullane** [comment]
Hmmm...wondering if the analyst really need the sort scenarios. A search by file name and applicant name is probably a lot more useful.

**Comment by JHarrietha-AOT** [comment]
@lmullane Thats a good point if an analyst is looking for a specific file they would just search for it. And we're organizing the queue in a default way with unopened at the top and then oldest to newest for opened requests. Which is based on the feedback from intact. So they don't feel necessary.

**Comment by lmullane** [comment]
@divyav-aot, this would be the next ticket to start on if you have time this Sprint. cc @abin-aot @mpilchar

**Comment by abin-aot** [comment]
"Will need a custom page created in Forms flow. Include Get from API." - This is subjected to discussion within the team, at this moment
